packages/tabs/src/tabs.tsx

```diff
@@ -50,22 +51,35 @@ export interface TabsProps
  * any DOM node.
  */
 export const Tabs = forwardRef<TabsProps, "div">((props, ref) => {
+  const [rootTabsElement, setRootTabsElement] = React.useState()
+
   const styles = useMultiStyleConfig("Tabs", props)
   const { children, className, ...rest } = omitThemingProps(props)
 
-  const { htmlProps, ...ctx } = useTabs(rest)
+  const { htmlProps, ...ctx } = useTabs({ ...rest, rootTabsElement })
   const context = React.useMemo(() => ctx, [ctx])
 
   const rootProps = omit(htmlProps as any, ["isFitted"])
 
+  const rootStyles: ChakraProps["__css"] = {
+    position: "relative",
+    ...styles.root,
+  }
+
+  const captureElement = React.useCallback((node) => {
+    if (node !== null) {
+      setRootTabsElement(node)
+    }
+  }, [])
```

Comment:
This is the ugliest part of this solution. It involves storing the element reference in context so that the tab indicator hook has access to the element to do positioning calculations:
https://github.com/chakra-ui/chakra-ui/pull/2778/files#diff-a67d97c904205ad37b1bd0c751d6d590e45b819f9246e455537c1f0bd040dfa6R515

Reply:
An alternative could be to get the tabs element by doing a [closest()](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/utils/src/dom-query.ts#L131) query to get the `.chakra-tabs`. 

This is not a very _react_ like approach, but it would simplify the code.

Reply:
I made the alternative approach in 81148db8c870f8989cb4017388906f030bbd9aa6

I think it's much simpler and a better approach